PennMobile/General/Protocols.swift

```diff
@@ -121,6 +121,37 @@ extension ShowsAlert where Self: UIViewController {
     }
 }
 
+protocol ShowsAlertForError : ShowsAlert {
+    func showAlertForError<T>(result: Result<T, NetworkingError>, title: String, success: @escaping (T) -> Void, noInternet: (() -> Void)?, parsingError: (() -> Void)?, serverError: (() -> Void)?, jsonError: (() -> Void)?, authenticationError: (() -> Void)?, other: (() -> Void)?)
+}
+
+extension ShowsAlertForError {
+    func showAlertForError<T>(result: Result<T, NetworkingError>, title: String, success: @escaping (T) -> Void, noInternet: (() -> Void)? = nil, parsingError: (() -> Void)? = nil, serverError: (() -> Void)? = nil, jsonError: (() -> Void)? = nil, authenticationError: (() -> Void)? = nil, other: (() -> Void)? = nil) {
+        switch result {
+        case .success(let content):
+            self.showAlert(withMsg: "Your \(title) has been refreshed.", title: "Refresh Complete!", completion: { success(content) })
+
+        case .failure(.noInternet):
+            self.showAlert(withMsg: "You appear to be offline.\nPlease try again later.", title: "Network Error", completion: noInternet)
+
+        case .failure(.parsingError):
+            self.showAlert(withMsg: "Penn's \(title) servers are currently not updating. We hope this will be fixed shortly", title: "Uh oh!", completion: parsingError)
+            
+        case .failure(.serverError):
+            self.showAlert(withMsg: "Penn's \(title) servers are currently not updating. We hope this will be fixed shortly", title: "Uh oh!", completion: serverError)
+            
+        case .failure(.jsonError):
+            self.showAlert(withMsg: "Penn's \(title) servers are currently not updating. We hope this will be fixed shortly", title: "Uh oh!", completion: jsonError)
```

Comment:
I wasn't too sure if it was the correct decision to make the messaging for internet, parsing, and json error all the same, informing the users that the issue wasn't on our end.